### DISCUSSION TITLE
Does dataloader destroy graph structure using mini-batch?

### DISCUSSION THREAD

**Discussion by jiaqian** [pyg-team/pytorch_geometric#5370] [Q&A]
Hi, I create a dataloader (e.g. `NeighborLoader`) by a big graph `G`(composed of some subgraphs), I am wondering does `batch` likely destroy the graph structures (for instance breaking some edges) with **small** batch size since I did not supply a list of subgraphs to dataloader? Any input is welcome, thanks.

**Comment by rusty1s** [comment]
Can you clarify? I am not totally sure I understand, sorry. If you apply the neighbor loader on a big supergraph containing many isolated subgraphs, these subgraphs will still be disconnected after sampling. Importantly, not all subgraphs will be complete, depending on the number of hops and neighbors you sample.
